src/android.ts

```diff
@@ -672,30 +698,24 @@ export const androidApi = <RunTarget extends SupportedRunTarget<'android'>>(
                 return;
             }
 
-            await this._internal.requireRoot('enabling a WireGuard tunnel');
-
-            const { stdout: appUser } = await adb(['shell', 'stat', '-c', '%U', '/data/data/com.wireguard.android']);
-            await adb([
-                'shell',
-                'su',
-                appUser,
-                '/bin/sh',
-                '-c',
-                `"echo -n '${config}' > /data/data/com.wireguard.android/files/${tunnelName}.conf"`,
+            const { stdout: appUser } = await adbRootShell(['stat', '-c', '%U', '/data/data/com.wireguard.android']);
+            await adbRootShell([
+                `'su ${appUser} /bin/sh -c "echo -n '${Buffer.from(config, 'utf-8').toString(
+                    'base64'
+                )}' | base64 -d > /data/data/com.wireguard.android/files/${tunnelName}.conf"'`,
             ]);
```

Comment:
Running this in a fresh Android 13 emulator, I get:

```
file:///home/benni/coding/JS/tweasel/appstraction/node_modules/andromatic/node_modules/execa/lib/error.js:59
		error = new Error(message);
		        ^

Error: Command failed with exit code 127: /home/benni/.cache/andromatic/platform-tools/adb shell 'su u0_a159 /bin/sh -c "echo -n 'LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURvVENDQW9tZ0F3SUJBZ0lHRHBxSmoxKzVNQTBHQ1NxR1NJYjNEUUVCQ3dVQU1DZ3hFakFRQmdOVkJBTU0KQ1cxcGRHMXdjbTk0ZVRFU01CQUdBMVVFQ2d3SmJXbDBiWEJ5YjNoNU1CNFhEVEl3TVRFeE5qQTROVEkwTWxvWApEVEl6TVRFeE9EQTROVEkwTWxvd0tERVNNQkFHQTFVRUF3d0piV2wwYlhCeWIzaDVNUkl3RUFZRFZRUUtEQWx0CmFYUnRjSEp2ZUhrd2dnRWlNQTBHQ1NxR1NJYjNEUUVCQVFVQUE0SUJEd0F3Z2dFS0FvSUJBUUNkZUppNU5lZlYKZWJPNkE0UGczbVF2ak8vblpIZDJ3QnRybVp2cDFCYjVOUk5rZHh1VXdJOW5EMGZTMk5pZTZCMEs2WVJmb3dJbwoxVFQzSFZGZzNIeXNlaUxuOTN0WFg3ZUt4N3hidnY3NWphc0pYRjMwZW1mMFZaZEZBT3doby9vblF6TjJBWUJDCkxUOWlOMm0rbkxPSFU4RkgyWUg5UXR1em81ZUZCOHdFY3FpTFRjU2E2M3A1elVBZUJzWkx0OFYzanFPejhXemMKOXExd044U0hsalQvTzlpa0k0SDdncHFuMWZsSVVFeFZIckxrNm9KWTZpeXhSTjlyK055UnlVL3BhTnYxcUZOMApMSWdSc0hjR1pQa2ZrOSt6SjZuTWg1aFFLYkNWaTdMN1RuM3E1NWJrVW1TaE42Zk0yY0tRK2xaRGhpSGpOQXFsCkEwR04xZVpIcDVsakFnTUJBQUdqZ2RBd2djMHdEd1lEVlIwVEFRSC9CQVV3QXdFQi96QVJCZ2xnaGtnQmh2aEMKQVFFRUJBTUNBZ1F3ZUFZRFZSMGxCSEV3YndZSUt3WUJCUVVIQXdFR0NDc0dBUVVGQndNQ0JnZ3JCZ0VGQlFjRApCQVlJS3dZQkJRVUhBd2dHQ2lzR0FRUUJnamNDQVJVR0Npc0dBUVFCZ2pjQ0FSWUdDaXNHQVFRQmdqY0tBd0VHCkNpc0dBUVFCZ2pjS0F3TUdDaXNHQVFRQmdqY0tBd1FHQ1dDR1NBR0crRUlFQVRBT0JnTlZIUThCQWY4RUJBTUMKQVFZd0hRWURWUjBPQkJZRUZLMDB4WkJwOTFWU0Q5aTRYeWo2SVVQdkJRa1BNQTBHQ1NxR1NJYjNEUUVCQ3dVQQpBNElCQVFCWVFjcWdPcEVtSGtNQ2tIR0VUYVBKY2RiZzVtZmhCL2xiSG9OQkJ1Q3NhT1ZsbmZQam11elBxeVFJCmprTmhBaGx2WUFNZXFUM21QQlZzVGdHd3dmejducmRNSnlQa05IdC9wZ01OaVozR29iaXRmdlFDZTFJanVTemIKSGZodVJQeUhSZlFwTkJDQ2Jlc1RualZ4eXl1WndLMFk5OXdWV25KcUVvamFja2haQTZYUWxPbTAwQVMxb2lyeApDVFkrQ0F1RTRsS1U5RWxVRzhYRFQ3VWpjeDlvT1VPakQ2S3lPeklVd04rSkRMZ3lDVGEzYXA4NkZseVlMN0dhCkpBcUxWV2lpUWR5RkpzQmgrcTNXUzhxcng0OXBmZlZvYTB2ZHEvVG11S0g5YjN4dVdiLzdOYitLOHpUd1ZOZTcKSENJL3ZPUTNLUWlWMEtDODAvSVZDYzZlZ2YwVAotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==' | base64 -d > /data/data/com.wireguard.android/files/appstraction.conf"'
/system/bin/sh: su u0_a159 /bin/sh -c "echo -n LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURvVENDQW9tZ0F3SUJBZ0lHRHBxSmoxKzVNQTBHQ1NxR1NJYjNEUUVCQ3dVQU1DZ3hFakFRQmdOVkJBTU0KQ1cxcGRHMXdjbTk0ZVRFU01CQUdBMVVFQ2d3SmJXbDBiWEJ5YjNoNU1CNFhEVEl3TVRFeE5qQTROVEkwTWxvWApEVEl6TVRFeE9EQTROVEkwTWxvd0tERVNNQkFHQTFVRUF3d0piV2wwYlhCeWIzaDVNUkl3RUFZRFZRUUtEQWx0CmFYUnRjSEp2ZUhrd2dnRWlNQTBHQ1NxR1NJYjNEUUVCQVFVQUE0SUJEd0F3Z2dFS0FvSUJBUUNkZUppNU5lZlYKZWJPNkE0UGczbVF2ak8vblpIZDJ3QnRybVp2cDFCYjVOUk5rZHh1VXdJOW5EMGZTMk5pZTZCMEs2WVJmb3dJbwoxVFQzSFZGZzNIeXNlaUxuOTN0WFg3ZUt4N3hidnY3NWphc0pYRjMwZW1mMFZaZEZBT3doby9vblF6TjJBWUJDCkxUOWlOMm0rbkxPSFU4RkgyWUg5UXR1em81ZUZCOHdFY3FpTFRjU2E2M3A1elVBZUJzWkx0OFYzanFPejhXemMKOXExd044U0hsalQvTzlpa0k0SDdncHFuMWZsSVVFeFZIckxrNm9KWTZpeXhSTjlyK055UnlVL3BhTnYxcUZOMApMSWdSc0hjR1pQa2ZrOSt6SjZuTWg1aFFLYkNWaTdMN1RuM3E1NWJrVW1TaE42Zk0yY0tRK2xaRGhpSGpOQXFsCkEwR04xZVpIcDVsakFnTUJBQUdqZ2RBd2djMHdEd1lEVlIwVEFRSC9CQVV3QXdFQi96QVJCZ2xnaGtnQmh2aEMKQVFFRUJBTUNBZ1F3ZUFZRFZSMGxCSEV3YndZSUt3WUJCUVVIQXdFR0NDc0dBUVVGQndNQ0JnZ3JCZ0VGQlFjRApCQVlJS3dZQkJRVUhBd2dHQ2lzR0FRUUJnamNDQVJVR0Npc0dBUVFCZ2pjQ0FSWUdDaXNHQVFRQmdqY0tBd0VHCkNpc0dBUVFCZ2pjS0F3TUdDaXNHQVFRQmdqY0tBd1FHQ1dDR1NBR0crRUlFQVRBT0JnTlZIUThCQWY4RUJBTUMKQVFZd0hRWURWUjBPQkJZRUZLMDB4WkJwOTFWU0Q5aTRYeWo2SVVQdkJRa1BNQTBHQ1NxR1NJYjNEUUVCQ3dVQQpBNElCQVFCWVFjcWdPcEVtSGtNQ2tIR0VUYVBKY2RiZzVtZmhCL2xiSG9OQkJ1Q3NhT1ZsbmZQam11elBxeVFJCmprTmhBaGx2WUFNZXFUM21QQlZzVGdHd3dmejducmRNSnlQa05IdC9wZ01OaVozR29iaXRmdlFDZTFJanVTemIKSGZodVJQeUhSZlFwTkJDQ2Jlc1RualZ4eXl1WndLMFk5OXdWV25KcUVvamFja2haQTZYUWxPbTAwQVMxb2lyeApDVFkrQ0F1RTRsS1U5RWxVRzhYRFQ3VWpjeDlvT1VPakQ2S3lPeklVd04rSkRMZ3lDVGEzYXA4NkZseVlMN0dhCkpBcUxWV2lpUWR5RkpzQmgrcTNXUzhxcng0OXBmZlZvYTB2ZHEvVG11S0g5YjN4dVdiLzdOYitLOHpUd1ZOZTcKSENJL3ZPUTNLUWlWMEtDODAvSVZDYzZlZ2YwVAotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg== | base64 -d > /data/data/com.wireguard.android/files/appstraction.conf": inaccessible or not found
    at makeError (file:///home/benni/coding/JS/tweasel/appstraction/node_modules/andromatic/node_modules/execa/lib/error.js:59:11)
    at handlePromise (file:///home/benni/coding/JS/tweasel/appstraction/node_modules/andromatic/node_modules/execa/index.js:124:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at Object.setProxy (/home/benni/coding/JS/tweasel/appstraction/src/android.ts:703:13)
    at <anonymous> (/home/benni/coding/JS/tweasel/appstraction/examples/android-emulator.ts:31:30) {
  shortMessage: `Command failed with exit code 127: /home/benni/.cache/andromatic/platform-tools/adb shell 'su u0_a159 /bin/sh -c "echo -n 'LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURvVENDQW9tZ0F3SUJBZ0lHRHBxSmoxKzVNQTBHQ1NxR1NJYjNEUUVCQ3dVQU1DZ3hFakFRQmdOVkJBTU0KQ1cxcGRHMXdjbTk0ZVRFU01CQUdBMVVFQ2d3SmJXbDBiWEJ5YjNoNU1CNFhEVEl3TVRFeE5qQTROVEkwTWxvWApEVEl6TVRFeE9EQTROVEkwTWxvd0tERVNNQkFHQTFVRUF3d0piV2wwYlhCeWIzaDVNUkl3RUFZRFZRUUtEQWx0CmFYUnRjSEp2ZUhrd2dnRWlNQTBHQ1NxR1NJYjNEUUVCQVFVQUE0SUJEd0F3Z2dFS0FvSUJBUUNkZUppNU5lZlYKZWJPNkE0UGczbVF2ak8vblpIZDJ3QnRybVp2cDFCYjVOUk5rZHh1VXdJOW5EMGZTMk5pZTZCMEs2WVJmb3dJbwoxVFQzSFZGZzNIeXNlaUxuOTN0WFg3ZUt4N3hidnY3NWphc0pYRjMwZW1mMFZaZEZBT3doby9vblF6TjJBWUJDCkxUOWlOMm0rbkxPSFU4RkgyWUg5UXR1em81ZUZCOHdFY3FpTFRjU2E2M3A1elVBZUJzWkx0OFYzanFPejhXemMKOXExd044U0hsalQvTzlpa0k0SDdncHFuMWZsSVVFeFZIckxrNm9KWTZpeXhSTjlyK055UnlVL3BhTnYxcUZOMApMSWdSc0hjR1pQa2ZrOSt6SjZuTWg1aFFLYkNWaTdMN1RuM3E1NWJrVW1TaE42Zk0yY0tRK2xaRGhpSGpOQXFsCkEwR04xZVpIcDVsakFnTUJBQUdqZ2RBd2djMHdEd1lEVlIwVEFRSC9CQVV3QXdFQi96QVJCZ2xnaGtnQmh2aEMKQVFFRUJBTUNBZ1F3ZUFZRFZSMGxCSEV3YndZSUt3WUJCUVVIQXdFR0NDc0dBUVVGQndNQ0JnZ3JCZ0VGQlFjRApCQVlJS3dZQkJRVUhBd2dHQ2lzR0FRUUJnamNDQVJVR0Npc0dBUVFCZ2pjQ0FSWUdDaXNHQVFRQmdqY0tBd0VHCkNpc0dBUVFCZ2pjS0F3TUdDaXNHQVFRQmdqY0tBd1FHQ1dDR1NBR0crRUlFQVRBT0JnTlZIUThCQWY4RUJBTUMKQVFZd0hRWURWUjBPQkJZRUZLMDB4WkJwOTFWU0Q5aTRYeWo2SVVQdkJRa1BNQTBHQ1NxR1NJYjNEUUVCQ3dVQQpBNElCQVFCWVFjcWdPcEVtSGtNQ2tIR0VUYVBKY2RiZzVtZmhCL2xiSG9OQkJ1Q3NhT1ZsbmZQam11elBxeVFJCmprTmhBaGx2WUFNZXFUM21QQlZzVGdHd3dmejducmRNSnlQa05IdC9wZ01OaVozR29iaXRmdlFDZTFJanVTemIKSGZodVJQeUhSZlFwTkJDQ2Jlc1RualZ4eXl1WndLMFk5OXdWV25KcUVvamFja2haQTZYUWxPbTAwQVMxb2lyeApDVFkrQ0F1RTRsS1U5RWxVRzhYRFQ3VWpjeDlvT1VPakQ2S3lPeklVd04rSkRMZ3lDVGEzYXA4NkZseVlMN0dhCkpBcUxWV2lpUWR5RkpzQmgrcTNXUzhxcng0OXBmZlZvYTB2ZHEvVG11S0g5YjN4dVdiLzdOYitLOHpUd1ZOZTcKSENJL3ZPUTNLUWlWMEtDODAvSVZDYzZlZ2YwVAotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==' | base64 -d > /data/data/com.wireguard.android/files/appstraction.conf"'`,
  command: `/home/benni/.cache/andromatic/platform-tools/adb shell 'su u0_a159 /bin/sh -c "echo -n 'LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURvVENDQW9tZ0F3SUJBZ0lHRHBxSmoxKzVNQTBHQ1NxR1NJYjNEUUVCQ3dVQU1DZ3hFakFRQmdOVkJBTU0KQ1cxcGRHMXdjbTk0ZVRFU01CQUdBMVVFQ2d3SmJXbDBiWEJ5YjNoNU1CNFhEVEl3TVRFeE5qQTROVEkwTWxvWApEVEl6TVRFeE9EQTROVEkwTWxvd0tERVNNQkFHQTFVRUF3d0piV2wwYlhCeWIzaDVNUkl3RUFZRFZRUUtEQWx0CmFYUnRjSEp2ZUhrd2dnRWlNQTBHQ1NxR1NJYjNEUUVCQVFVQUE0SUJEd0F3Z2dFS0FvSUJBUUNkZUppNU5lZlYKZWJPNkE0UGczbVF2ak8vblpIZDJ3QnRybVp2cDFCYjVOUk5rZHh1VXdJOW5EMGZTMk5pZTZCMEs2WVJmb3dJbwoxVFQzSFZGZzNIeXNlaUxuOTN0WFg3ZUt4N3hidnY3NWphc0pYRjMwZW1mMFZaZEZBT3doby9vblF6TjJBWUJDCkxUOWlOMm0rbkxPSFU4RkgyWUg5UXR1em81ZUZCOHdFY3FpTFRjU2E2M3A1elVBZUJzWkx0OFYzanFPejhXemMKOXExd044U0hsalQvTzlpa0k0SDdncHFuMWZsSVVFeFZIckxrNm9KWTZpeXhSTjlyK055UnlVL3BhTnYxcUZOMApMSWdSc0hjR1pQa2ZrOSt6SjZuTWg1aFFLYkNWaTdMN1RuM3E1NWJrVW1TaE42Zk0yY0tRK2xaRGhpSGpOQXFsCkEwR04xZVpIcDVsakFnTUJBQUdqZ2RBd2djMHdEd1lEVlIwVEFRSC9CQVV3QXdFQi96QVJCZ2xnaGtnQmh2aEMKQVFFRUJBTUNBZ1F3ZUFZRFZSMGxCSEV3YndZSUt3WUJCUVVIQXdFR0NDc0dBUVVGQndNQ0JnZ3JCZ0VGQlFjRApCQVlJS3dZQkJRVUhBd2dHQ2lzR0FRUUJnamNDQVJVR0Npc0dBUVFCZ2pjQ0FSWUdDaXNHQVFRQmdqY0tBd0VHCkNpc0dBUVFCZ2pjS0F3TUdDaXNHQVFRQmdqY0tBd1FHQ1dDR1NBR0crRUlFQVRBT0JnTlZIUThCQWY4RUJBTUMKQVFZd0hRWURWUjBPQkJZRUZLMDB4WkJwOTFWU0Q5aTRYeWo2SVVQdkJRa1BNQTBHQ1NxR1NJYjNEUUVCQ3dVQQpBNElCQVFCWVFjcWdPcEVtSGtNQ2tIR0VUYVBKY2RiZzVtZmhCL2xiSG9OQkJ1Q3NhT1ZsbmZQam11elBxeVFJCmprTmhBaGx2WUFNZXFUM21QQlZzVGdHd3dmejducmRNSnlQa05IdC9wZ01OaVozR29iaXRmdlFDZTFJanVTemIKSGZodVJQeUhSZlFwTkJDQ2Jlc1RualZ4eXl1WndLMFk5OXdWV25KcUVvamFja2haQTZYUWxPbTAwQVMxb2lyeApDVFkrQ0F1RTRsS1U5RWxVRzhYRFQ3VWpjeDlvT1VPakQ2S3lPeklVd04rSkRMZ3lDVGEzYXA4NkZseVlMN0dhCkpBcUxWV2lpUWR5RkpzQmgrcTNXUzhxcng0OXBmZlZvYTB2ZHEvVG11S0g5YjN4dVdiLzdOYitLOHpUd1ZOZTcKSENJL3ZPUTNLUWlWMEtDODAvSVZDYzZlZ2YwVAotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==' | base64 -d > /data/data/com.wireguard.android/files/appstraction.conf"'`,
  escapedCommand: `"/home/benni/.cache/andromatic/platform-tools/adb" shell "'su u0_a159 /bin/sh -c \\"echo -n 'LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURvVENDQW9tZ0F3SUJBZ0lHRHBxSmoxKzVNQTBHQ1NxR1NJYjNEUUVCQ3dVQU1DZ3hFakFRQmdOVkJBTU0KQ1cxcGRHMXdjbTk0ZVRFU01CQUdBMVVFQ2d3SmJXbDBiWEJ5YjNoNU1CNFhEVEl3TVRFeE5qQTROVEkwTWxvWApEVEl6TVRFeE9EQTROVEkwTWxvd0tERVNNQkFHQTFVRUF3d0piV2wwYlhCeWIzaDVNUkl3RUFZRFZRUUtEQWx0CmFYUnRjSEp2ZUhrd2dnRWlNQTBHQ1NxR1NJYjNEUUVCQVFVQUE0SUJEd0F3Z2dFS0FvSUJBUUNkZUppNU5lZlYKZWJPNkE0UGczbVF2ak8vblpIZDJ3QnRybVp2cDFCYjVOUk5rZHh1VXdJOW5EMGZTMk5pZTZCMEs2WVJmb3dJbwoxVFQzSFZGZzNIeXNlaUxuOTN0WFg3ZUt4N3hidnY3NWphc0pYRjMwZW1mMFZaZEZBT3doby9vblF6TjJBWUJDCkxUOWlOMm0rbkxPSFU4RkgyWUg5UXR1em81ZUZCOHdFY3FpTFRjU2E2M3A1elVBZUJzWkx0OFYzanFPejhXemMKOXExd044U0hsalQvTzlpa0k0SDdncHFuMWZsSVVFeFZIckxrNm9KWTZpeXhSTjlyK055UnlVL3BhTnYxcUZOMApMSWdSc0hjR1pQa2ZrOSt6SjZuTWg1aFFLYkNWaTdMN1RuM3E1NWJrVW1TaE42Zk0yY0tRK2xaRGhpSGpOQXFsCkEwR04xZVpIcDVsakFnTUJBQUdqZ2RBd2djMHdEd1lEVlIwVEFRSC9CQVV3QXdFQi96QVJCZ2xnaGtnQmh2aEMKQVFFRUJBTUNBZ1F3ZUFZRFZSMGxCSEV3YndZSUt3WUJCUVVIQXdFR0NDc0dBUVVGQndNQ0JnZ3JCZ0VGQlFjRApCQVlJS3dZQkJRVUhBd2dHQ2lzR0FRUUJnamNDQVJVR0Npc0dBUVFCZ2pjQ0FSWUdDaXNHQVFRQmdqY0tBd0VHCkNpc0dBUVFCZ2pjS0F3TUdDaXNHQVFRQmdqY0tBd1FHQ1dDR1NBR0crRUlFQVRBT0JnTlZIUThCQWY4RUJBTUMKQVFZd0hRWURWUjBPQkJZRUZLMDB4WkJwOTFWU0Q5aTRYeWo2SVVQdkJRa1BNQTBHQ1NxR1NJYjNEUUVCQ3dVQQpBNElCQVFCWVFjcWdPcEVtSGtNQ2tIR0VUYVBKY2RiZzVtZmhCL2xiSG9OQkJ1Q3NhT1ZsbmZQam11elBxeVFJCmprTmhBaGx2WUFNZXFUM21QQlZzVGdHd3dmejducmRNSnlQa05IdC9wZ01OaVozR29iaXRmdlFDZTFJanVTemIKSGZodVJQeUhSZlFwTkJDQ2Jlc1RualZ4eXl1WndLMFk5OXdWV25KcUVvamFja2haQTZYUWxPbTAwQVMxb2lyeApDVFkrQ0F1RTRsS1U5RWxVRzhYRFQ3VWpjeDlvT1VPakQ2S3lPeklVd04rSkRMZ3lDVGEzYXA4NkZseVlMN0dhCkpBcUxWV2lpUWR5RkpzQmgrcTNXUzhxcng0OXBmZlZvYTB2ZHEvVG11S0g5YjN4dVdiLzdOYitLOHpUd1ZOZTcKSENJL3ZPUTNLUWlWMEtDODAvSVZDYzZlZ2YwVAotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==' | base64 -d > /data/data/com.wireguard.android/files/appstraction.conf\\"'"`,
  exitCode: 127,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: '/system/bin/sh: su u0_a159 /bin/sh -c "echo -n LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURvVENDQW9tZ0F3SUJBZ0lHRHBxSmoxKzVNQTBHQ1NxR1NJYjNEUUVCQ3dVQU1DZ3hFakFRQmdOVkJBTU0KQ1cxcGRHMXdjbTk0ZVRFU01CQUdBMVVFQ2d3SmJXbDBiWEJ5YjNoNU1CNFhEVEl3TVRFeE5qQTROVEkwTWxvWApEVEl6TVRFeE9EQTROVEkwTWxvd0tERVNNQkFHQTFVRUF3d0piV2wwYlhCeWIzaDVNUkl3RUFZRFZRUUtEQWx0CmFYUnRjSEp2ZUhrd2dnRWlNQTBHQ1NxR1NJYjNEUUVCQVFVQUE0SUJEd0F3Z2dFS0FvSUJBUUNkZUppNU5lZlYKZWJPNkE0UGczbVF2ak8vblpIZDJ3QnRybVp2cDFCYjVOUk5rZHh1VXdJOW5EMGZTMk5pZTZCMEs2WVJmb3dJbwoxVFQzSFZGZzNIeXNlaUxuOTN0WFg3ZUt4N3hidnY3NWphc0pYRjMwZW1mMFZaZEZBT3doby9vblF6TjJBWUJDCkxUOWlOMm0rbkxPSFU4RkgyWUg5UXR1em81ZUZCOHdFY3FpTFRjU2E2M3A1elVBZUJzWkx0OFYzanFPejhXemMKOXExd044U0hsalQvTzlpa0k0SDdncHFuMWZsSVVFeFZIckxrNm9KWTZpeXhSTjlyK055UnlVL3BhTnYxcUZOMApMSWdSc0hjR1pQa2ZrOSt6SjZuTWg1aFFLYkNWaTdMN1RuM3E1NWJrVW1TaE42Zk0yY0tRK2xaRGhpSGpOQXFsCkEwR04xZVpIcDVsakFnTUJBQUdqZ2RBd2djMHdEd1lEVlIwVEFRSC9CQVV3QXdFQi96QVJCZ2xnaGtnQmh2aEMKQVFFRUJBTUNBZ1F3ZUFZRFZSMGxCSEV3YndZSUt3WUJCUVVIQXdFR0NDc0dBUVVGQndNQ0JnZ3JCZ0VGQlFjRApCQVlJS3dZQkJRVUhBd2dHQ2lzR0FRUUJnamNDQVJVR0Npc0dBUVFCZ2pjQ0FSWUdDaXNHQVFRQmdqY0tBd0VHCkNpc0dBUVFCZ2pjS0F3TUdDaXNHQVFRQmdqY0tBd1FHQ1dDR1NBR0crRUlFQVRBT0JnTlZIUThCQWY4RUJBTUMKQVFZd0hRWURWUjBPQkJZRUZLMDB4WkJwOTFWU0Q5aTRYeWo2SVVQdkJRa1BNQTBHQ1NxR1NJYjNEUUVCQ3dVQQpBNElCQVFCWVFjcWdPcEVtSGtNQ2tIR0VUYVBKY2RiZzVtZmhCL2xiSG9OQkJ1Q3NhT1ZsbmZQam11elBxeVFJCmprTmhBaGx2WUFNZXFUM21QQlZzVGdHd3dmejducmRNSnlQa05IdC9wZ01OaVozR29iaXRmdlFDZTFJanVTemIKSGZodVJQeUhSZlFwTkJDQ2Jlc1RualZ4eXl1WndLMFk5OXdWV25KcUVvamFja2haQTZYUWxPbTAwQVMxb2lyeApDVFkrQ0F1RTRsS1U5RWxVRzhYRFQ3VWpjeDlvT1VPakQ2S3lPeklVd04rSkRMZ3lDVGEzYXA4NkZseVlMN0dhCkpBcUxWV2lpUWR5RkpzQmgrcTNXUzhxcng0OXBmZlZvYTB2ZHEvVG11S0g5YjN4dVdiLzdOYitLOHpUd1ZOZTcKSENJL3ZPUTNLUWlWMEtDODAvSVZDYzZlZ2YwVAotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg== | base64 -d > /data/data/com.wireguard.android/files/appstraction.conf": inaccessible or not found',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
```

Reply:
Curiously, if I run the command[^1] manually in a terminal, it does work.

[^1]: `/home/benni/.cache/andromatic/platform-tools/adb shell 'su u0_a159 /bin/sh -c "echo -n 'LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURvVENDQW9tZ0F3SUJBZ0lHRHBxSmoxKzVNQTBHQ1NxR1NJYjNEUUVCQ3dVQU1DZ3hFakFRQmdOVkJBTU0KQ1cxcGRHMXdjbTk0ZVRFU01CQUdBMVVFQ2d3SmJXbDBiWEJ5YjNoNU1CNFhEVEl3TVRFeE5qQTROVEkwTWxvWApEVEl6TVRFeE9EQTROVEkwTWxvd0tERVNNQkFHQTFVRUF3d0piV2wwYlhCeWIzaDVNUkl3RUFZRFZRUUtEQWx0CmFYUnRjSEp2ZUhrd2dnRWlNQTBHQ1NxR1NJYjNEUUVCQVFVQUE0SUJEd0F3Z2dFS0FvSUJBUUNkZUppNU5lZlYKZWJPNkE0UGczbVF2ak8vblpIZDJ3QnRybVp2cDFCYjVOUk5rZHh1VXdJOW5EMGZTMk5pZTZCMEs2WVJmb3dJbwoxVFQzSFZGZzNIeXNlaUxuOTN0WFg3ZUt4N3hidnY3NWphc0pYRjMwZW1mMFZaZEZBT3doby9vblF6TjJBWUJDCkxUOWlOMm0rbkxPSFU4RkgyWUg5UXR1em81ZUZCOHdFY3FpTFRjU2E2M3A1elVBZUJzWkx0OFYzanFPejhXemMKOXExd044U0hsalQvTzlpa0k0SDdncHFuMWZsSVVFeFZIckxrNm9KWTZpeXhSTjlyK055UnlVL3BhTnYxcUZOMApMSWdSc0hjR1pQa2ZrOSt6SjZuTWg1aFFLYkNWaTdMN1RuM3E1NWJrVW1TaE42Zk0yY0tRK2xaRGhpSGpOQXFsCkEwR04xZVpIcDVsakFnTUJBQUdqZ2RBd2djMHdEd1lEVlIwVEFRSC9CQVV3QXdFQi96QVJCZ2xnaGtnQmh2aEMKQVFFRUJBTUNBZ1F3ZUFZRFZSMGxCSEV3YndZSUt3WUJCUVVIQXdFR0NDc0dBUVVGQndNQ0JnZ3JCZ0VGQlFjRApCQVlJS3dZQkJRVUhBd2dHQ2lzR0FRUUJnamNDQVJVR0Npc0dBUVFCZ2pjQ0FSWUdDaXNHQVFRQmdqY0tBd0VHCkNpc0dBUVFCZ2pjS0F3TUdDaXNHQVFRQmdqY0tBd1FHQ1dDR1NBR0crRUlFQVRBT0JnTlZIUThCQWY4RUJBTUMKQVFZd0hRWURWUjBPQkJZRUZLMDB4WkJwOTFWU0Q5aTRYeWo2SVVQdkJRa1BNQTBHQ1NxR1NJYjNEUUVCQ3dVQQpBNElCQVFCWVFjcWdPcEVtSGtNQ2tIR0VUYVBKY2RiZzVtZmhCL2xiSG9OQkJ1Q3NhT1ZsbmZQam11elBxeVFJCmprTmhBaGx2WUFNZXFUM21QQlZzVGdHd3dmejducmRNSnlQa05IdC9wZ01OaVozR29iaXRmdlFDZTFJanVTemIKSGZodVJQeUhSZlFwTkJDQ2Jlc1RualZ4eXl1WndLMFk5OXdWV25KcUVvamFja2haQTZYUWxPbTAwQVMxb2lyeApDVFkrQ0F1RTRsS1U5RWxVRzhYRFQ3VWpjeDlvT1VPakQ2S3lPeklVd04rSkRMZ3lDVGEzYXA4NkZseVlMN0dhCkpBcUxWV2lpUWR5RkpzQmgrcTNXUzhxcng0OXBmZlZvYTB2ZHEvVG11S0g5YjN4dVdiLzdOYitLOHpUd1ZOZTcKSENJL3ZPUTNLUWlWMEtDODAvSVZDYzZlZ2YwVAotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==' | base64 -d > /data/data/com.wireguard.android/files/appstraction.conf"'`

Reply:
There's also a bug in the emulator example script (introduced by me): The WireGuard config is in `process.argv[6]`.

I was a bit surprised at how long the WireGuard config was supposedly. Correct command (but still failing):

```sh
/home/benni/.cache/andromatic/platform-tools/adb shell 'su u0_a159 /bin/sh -c "echo -n 'W0ludGVyZmFjZV0KUHJpdmF0ZUtleSA9IEVGUlRMSGxhRmZFOWRYbnU0VHJkRTFvc2JUa0ZFMFMxMTA5Z2FMaUd2MXM9CkFkZHJlc3MgPSAxMC4wLjAuMS8zMgpETlMgPSAxMC4wLjAuNTMKCltQZWVyXQpQdWJsaWNLZXkgPSBHYmd4b3o4YiswOEJneFlBdmhlTTZPbUZUMDlhYVNReHRVUldZU2htOWkwPQpBbGxvd2VkSVBzID0gMC4wLjAuMC8wCkVuZHBvaW50ID0gMTAuMC4wLjY4OjUxODIwCg==' | base64 -d > /data/data/com.wireguard.android/files/appstraction.conf"'
```

Reply:
Other than this, it does also work in an emulator.

Reply:
Oh, this is annoying: Android uses a non-POSIX `su` (https://stackoverflow.com/questions/55506205/android-shell-su-doesnt-have-option-c)

Reply:
Wow. This took forever to fix. Why does Android not respect POSIX :(((